tests/integration/test_agent.py

```diff
@@ -123,16 +125,72 @@ async def test_jenkins_k8s_agent_relation(
 async def test_jenkins_k8s_agent_relation_removed(
     model: Model,
     jenkins_k8s_agent_related: Application,
-    jenkins_k8s_agent: Application,
+    jenkins_multi_k8s_agents: Application,
     jenkins_client: jenkinsapi.jenkins.Jenkins,
 ):
     """
     arrange: given a jenkins server charm related to jenkins-k8s-agent charm via agent relation.
     act: when the relation is removed.
     assert: no agent nodes remain registered.
     """
-    await jenkins_k8s_agent_related.remove_relation("agent", f"{jenkins_k8s_agent.name}:agent")
-    await model.wait_for_idle()
+    await jenkins_k8s_agent_related.remove_relation(
+        state.AGENT_RELATION, f"{jenkins_multi_k8s_agents.name}:{state.AGENT_RELATION}"
+    )
+    await model.wait_for_idle(apps=[jenkins_k8s_agent_related.name, jenkins_multi_k8s_agents.name])
+
+    nodes = jenkins_client.get_nodes()
+    assert not any((jenkins_multi_k8s_agents.name in key for key in nodes.keys()))
+
+
+async def test_jenkins_machine_agent_relation(
+    model: Model,
+    application: Application,
+    jenkins_multi_machine_agents: Application,
+    jenkins_client: jenkinsapi.jenkins.Jenkins,
+    gen_jenkins_test_job_xml: typing.Callable[[str], str],
+):
+    """
+    arrange: given a cross controller cross model jenkins machine agent with an offer.
+    act: when the relation is setup through the offer.
+    assert: the relation succeeds and the agent is able to run jobs successfully.
+    """
+    machine_model: Model = jenkins_multi_machine_agents.model
+    await application.relate(
+        state.AGENT_RELATION,
+        f"localhost:admin/{machine_model.name}.{jenkins_multi_machine_agents.name}",
+    )
+    await model.wait_for_idle(apps=[application.name], status="active")
```

Comment:
Minor/no need to change: There is a wait_for_active parameter that can be used if you want all the units to be active.

Reply:
changed! quite neat! thank you for the review!